backend/src/main/java/com/mcmasterbaja/analyzer/ConstantAdderAnalyzer.java

```diff
@@ -0,0 +1,77 @@
+package com.mcmasterbaja.analyzer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.mcmasterbaja.readwrite.CSVReader;
+import com.mcmasterbaja.readwrite.CSVWriter;
+import com.mcmasterbaja.readwrite.Reader;
+import com.mcmasterbaja.readwrite.Writer;
+
+public class ConstantAdderAnalyzer extends Analyzer{
+    private final double a;
+    private final double b;
+    private final double c;
+    private final double d;
+
+   public ConstantAdderAnalyzer(String[] inputFiles, String[] inputColumns, String[] outputFiles, double a, double b, double c, double d) {
+       super(inputFiles, inputColumns, outputFiles);
+       this.a = a;
+       this.b = b;
+       this.c = c;
+       this.d = d;
+   }
+  public void analyze() {
+
+    System.out.println("Add a constant value to a file named"+ super.inputFiles[0]+ " to make " + super.outputFiles[0]);
+    Reader r = new CSVReader(super.inputFiles[0]);
+    Writer w = new CSVWriter(super.outputFiles[0]);
```

Comment:
I may have misled you a bit on this front, something we're looking to use going forward is to move away from using our custom CSVReader and CSVWriter and move to openCSV library's stuff. There's a whole PR open about it right now you can see here https://github.com/McMaster-Baja-Racing/Better-Data-Viewer/pull/117/files which shows a great example of changing from one to the other.

Reply:
Will need some guidance on which parts can stay vs need to be changed

Reply:
In this [PR](https://github.com/McMaster-Baja-Racing/Better-Data-Viewer/pull/135/files#diff-bfabd33cfd2988b7b25e7a2a4b452971138a48785086fae65b0f5cec7225fd33) you can see some examples like in the image, where I swap out the CSVReader and CSVWriter (Which we define locally) for the ones that OpenCSV has. By consequence, we should also be reading files line by line, rather than loading them all at once. So instead of r.read() and then looping on it, it should loop while there exists a next line, and then performs the operations and writes them. Something like the second image here

![image](https://github.com/user-attachments/assets/9b9ec085-65c3-4453-8fb0-7438837d4f26)

![image](https://github.com/user-attachments/assets/bc6fd2f3-1666-43e6-bd05-074bad2ff572)

Reply:
good now